File: Code/new_fitervari_backend/src/main/java/at/htl/model/ExerciseHistory.java
package at.htl.model;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "exercise_history")
public class ExerciseHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "exercise_id")
    private int exerciseId;

    @OneToMany(
            mappedBy = "exerciseHistoryId",
            cascade = CascadeType.ALL,
            orphanRemoval = true,
            fetch = FetchType.EAGER
    )
    private List<SetHistory> setHistories;

    //region GetterSetter
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getExerciseId() {
        return exerciseId;
    }

    public void setExerciseId(int exerciseId) {
        this.exerciseId = exerciseId;
    }

    public List<SetHistory> getSetHistories() {
        return setHistories;
    }

    public void setSetHistories(List<SetHistory> setHistories) {
        this.setHistories = setHistories;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/ExerciseHistory.kt
package at.htl_leonding.model

import io.quarkus.hibernate.orm.panache.PanacheEntity
import javax.json.bind.annotation.JsonbTransient
import javax.persistence.*
import kotlin.collections.Set

@Entity
@Table(name = "exercise_history")
class ExerciseHistory(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @OneToMany(
        mappedBy = "exercise_history_id",
        cascade = arrayOf(CascadeType.ALL),
        orphanRemoval = true,
        fetch = FetchType.EAGER
    )
    var setHistories: MutableList<SetHistory> = mutableListOf(),
    var exercise_id: Int
) {
    fun copyValues(other: ExerciseHistory) {
        this.setHistories = other.setHistories
        this.exercise_id = exercise_id
    }
}